addons/pos_self_order/static/src/app/models/product_custom_attribute.js

```diff
@@ -0,0 +1,21 @@
+/** @odoo-module **/
+import { Reactive } from "@web/core/utils/reactive";
+
+export class ProductCustomAttribute extends Reactive {
```

Comment:
Reuse the class you created from point_of_sale module. To be able to reuse, declare it in a different file. And load that file here in pos_self_order module.